### DISCUSSION TITLE
JavaScript Actions in Alpine containers are only supported on x64 Linux runners. Detected Linux Arm64

### DISCUSSION THREAD

**Discussion by ygini** [community/community#53407] [Actions]
### Select Topic Area

Product Feedback

### Body

Currently JavaScript based GitHub actions are limited to x64 architecture.

Our current project require delivery for both x64 and ARM64. With have runners on both architectures for performance reasons.

Our workflow is a success of matrix of arch, for the stages build / create container / merge in a manifest.

And we use the artifacts system to pass the build result from the build stage to the create container one.

We can't create the container at the build stage because it's a specific ci-container we use with no support for docker in docker.

Moving ARM64 to a first class citizen in GitHub action could really be nice for 2023…

**Comment by mycodedoesnotcompile2** [comment]
2025 and this is still an issue:
https://github.com/actions/upload-artifact/issues/739
